packages/monitor-v2/src/bot-oo-v3/SettleAssertions.ts

```diff
@@ -0,0 +1,59 @@
+import { getContractInstanceWithProvider, Logger, MonitoringParams, OptimisticOracleV3Ethers } from "./common";
+import { logSettleAssertion } from "./BotLogger";
+import { paginatedEventQuery } from "../utils/EventUtils";
+import {
+  AssertionSettledEvent,
+  AssertionMadeEvent,
+} from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/OptimisticOracleV3";
+
+export async function settleAssertions(logger: typeof Logger, params: MonitoringParams): Promise<void> {
+  const oo = await getContractInstanceWithProvider<OptimisticOracleV3Ethers>("OptimisticOracleV3", params.provider);
+
+  const currentBlockNumber = await params.provider.getBlockNumber();
+
+  const assertions = await paginatedEventQuery<AssertionMadeEvent>(oo, oo.filters.AssertionMade(), {
+    fromBlock: 0,
+    toBlock: currentBlockNumber,
```

Comment:
I think to make this paginate, you'll need to specify a maxBlockLookback value in the search config -- that will tell the method how many blocks it can query at a time for that chain.

Reply:
Thanks for your comments @mrice32 . I think I misunderstood the functioning of the `paginatedEventQuery` function.
I've added some changes to the script and new params `WARMING_UP_BLOCK_LOOKBACK, BLOCK_LOOKBACK, MAX_BLOCK_LOOKBACK`

Please let me know if you think this make sense!